### ISSUE TITLE
Fix prerelease version naming of building packages of PSCore6

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#5846]
The version should have been `6.0.0~beta.1` instead of `6.0.0-beta.1`.  The final version will be `6.0.0-1` which is alphabetically behind all the current prerelease versions so `apt install powershell` installs rc2 instead of final.

The convention for prerelease versioning is taken from http://gazebosim.org/tutorials?tut=install_unstable&cat=install

same for RPMs https://en.opensuse.org/openSUSE:Package_naming_guidelines

**Comment by chuanjiao10** [comment]
PSCore6 - Won't install in Ubuntu thru the apt Update/Upgrade   ----centos7 had same problem!

Steps to reproduce
------------------
yum update powershell 
#return no packages for update


yum remove powershell
yum clean all
yum install powershell
#return           install    6.0.0_rc.2-1.rhel.7  y/n ?
  
Big problem!

**Comment by SteveL-MSFT** [comment]
We're well aware of this and working on a solution

**Comment by chuanjiao10** [comment]
1 The following statement, can only install rc.2
```
I already had PowerShell Core on my Linux machine, and it’s not updating. How do I get the latest version?
We’re currently working through a known issue where powershell-6.0.0 is viewed by package managers as a lower version than powershell-6.0.0-rc2.
In the meantime, a clean install will get you the latest version:
sudo apt remove powershell && sudo apt-get install powershell
sudo yum remove powershell && sudo yum install powershell
```


2 i installed rpm from github,then run that command , Still let me upgrade from 6.0 to rc. 2.
```
--> 正在检查事务
---> 软件包 powershell.x86_64.0.6.0.0-1.rhel.7 将被 升级
---> 软件包 powershell.x86_64.0.6.0.0_rc.2-1.rhel.7 将被 更新
--> 解决依赖关系完成

依赖关系解决

==================================================================================================================================================
 Package                      架构                     版本                                   源                                             大小
==================================================================================================================================================
正在更新:
 powershell                   x86_64                   6.0.0_rc.2-1.rhel.7                    packages-microsoft-com-prod                    49 M

事务概要
==================================================================================================================================================
升级  1 软件包

总下载量：49 M
Is this ok [y/d/N]: n

```
3 I think it is big problem.I suggest quick fixes.Nobody cares about this problem?
4 I think the repair is simple, only need to modify the file name in https://packages.mircosoft.com.

**Comment by SteveL-MSFT** [comment]
@chuanjiao10 it's not that simple.  We've already removed the prerelease pkgs from packages.microsoft.com, but there is caching going on.  Also for the RPM, if you have RC2 installed, it thinks that's newer so by default doesn't allow you to install 6.0.0.  Current plan is to re-release 6.0.0 as 6.0.1 which should resolve this issue.

**Comment by chuanjiao10** [comment]
i tested with yum just now,it fixed! now good!
thank you!

**Comment by iSazonov** [comment]
I propose to consider [GitVersion](http://gitversion.readthedocs.io/en/latest/more-info/version-increments/)

**Comment by iSazonov** [comment]
From above link the highest priority recommendation is:
>Some upstreams assign their prereleases sortable numbers (that no other normal release will obtain). For example, sssd-1.8.0beta2.tar.gz is defined to be “1.7.92”. Use Version: 1.7.92 (in .spec file).

So for PowerShell Core 6.1.0 GA we could use the scheme: "unreal" version starting with 10000 - etc. 6.0.1000. We could consider 10001 - alpha1, 10002 - alfa2, 20001 - beta1, 20002 - beta2, 30001 - rc1. It allow continue use information suffixes for all package managers. 
It seems the schema resolve all problems. We could easily implement it.

**Comment by iSazonov** [comment]
Fixed in #6318
